File: leetcode/0169-majority-element/solution.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        h ={


        }
        for i in nums:
            if i in h:
                h[i]+=1
            else:
                h[i] =1
        for i in nums:
            if h[i] > len(nums)/2:
                return i
                
        

File: leetcode/0169-majority-element/solution.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var majorityElement = function(nums) {
    let n = 0
    let cand = null
    for(let num of nums){
        if(n===0){
            cand = num
        }
        n+=(cand===num)?1:-1
    }
    return cand
};
